test/functionalTest/cases/1259_new_json_for_v2/put_attribute_value_type_check.test

```diff
@@ -0,0 +1,109 @@
+# Copyright 2016 Telefonica Investigacion y Desarrollo, S.A.U
+#
+# This file is part of Orion Context Broker.
+#
+# Orion Context Broker is free software: you can redistribute it and/or
+# modify it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# Orion Context Broker is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+#
+# For those usages not covered by this license please contact with
+# iot_support at tid dot es
+
+# VALGRIND_READY - to mark the test ready for valgrindTestSuite.sh
+
+--NAME--
+PUT Attribute Value with attribute type check
+
+--SHELL-INIT--
+dbInit CB
+brokerStart CB 0-255
+
+--SHELL--
+
+#
+# 01. Create entity E2, with attribute A1 of type 'string'
+# 02. PUT /v2/entities/E2/attrs/A1/value with TEXT payload 'abc'
+# 03. GET /v2/entities/E2 to see A1 of type 'string' with value 'abc'
+#
+
+echo "01. Create entity E2, with attribute A1 of type 'string'"
+echo "========================================================"
+payload='{
+  "id": "E2",
+  "type": "T2",
+  "A1": { 
+    "value": 1,
+    "type": "string"
+  }
+}'
+orionCurl --url /v2/entities --payload "$payload"
+echo
+echo
+
+
+echo "02. PUT /v2/entities/E2/attrs/A1/value with TEXT payload 'abc'"
+echo "=============================================================="
+payload='"abc"'
+orionCurl --url /v2/entities/E2/attrs/A1/value --payload "$payload" -X PUT --in text/plain --out application/json
```

Comment:
I understand that `--out` maps to the Accept header at curl level. In that case, you don't need it, as this operation hasn't any payload in the response.